src/components/network/network-table.js

```diff
@@ -1234,6 +1930,286 @@ const NetworkTable = (props) => {
+    }
+
+    function union(a, b) {
+        return [...a, ...not(b, a)];
+    }
+
+    const handleToggle = (value) => () => {
+        const currentIndex = checked.indexOf(value);
+        const newChecked = [...checked];
+
+        if (currentIndex === -1) {
+            newChecked.push(value);
+        } else {
+            newChecked.splice(currentIndex, 1);
+        }
+
+        setChecked(newChecked);
+    };
+
+    const numberOfChecked = (items) => intersection(checked, items).length;
+
+    const handleToggleAll = (items) => () => {
+        if (numberOfChecked(items) === items.length) {
+            setChecked(not(checked, items));
+        } else {
+            setChecked(union(checked, items));
+        }
+    };
+
+    const checkListColumnsNames = () => {
+        return (
+            <List>
+                <ListItem
+                    className={classes.checkboxSelectAll}
+                    onClick={handleToggleAll(listColumnsNames)}
+                >
+                    <Checkbox
+                        checked={
+                            numberOfChecked(listColumnsNames) ===
+                                listColumnsNames.length &&
+                            listColumnsNames.length !== 0
+                        }
+                        indeterminate={
+                            numberOfChecked(listColumnsNames) !==
+                                listColumnsNames.length &&
+                            numberOfChecked(listColumnsNames) !== 0
+                        }
+                        disabled={listColumnsNames.length === 0}
+                        color="primary"
+                    />
+                    <FormattedMessage id="CheckAll" />
+                </ListItem>
+                {listColumnsNames.map((value, index) => (
+                    <List
+                        className="dragHandle"
+                        key={index}
+                        style={{ padding: '0' }}
+                    >
+                        <ListItem
+                            key={index}
+                            className={classes.checkboxItem}
+                            onClick={handleToggle(value)}
+                            style={{ padding: '0 16px' }}
+                        >
+                            <ListItemIcon>
+                                <Checkbox
+                                    checked={checked.indexOf(value) !== -1}
+                                    color="primary"
+                                />
+                            </ListItemIcon>
+                            <ListItemText primary={`${value}`} />
+                        </ListItem>
+                    </List>
+                ))}
+            </List>
+        );
+    };
+
+    const onClickCallFunction = (index) => {
+        let list = '';
+        if (props.network) {
+            setSelectedListName([]);
+            switch (index) {
+                case 0:
```

Comment:
We should refactor this to use the data defining the column, instead of redoing a switch/case here

Reply:
Done